src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -64,6 +64,133 @@ void FieldSpecificationManager::expandObjectCatalogs()
   }
 }
 
+void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) const
+{
+  // loop over all the FieldSpecification of the XML file
+  this->forSubGroups< FieldSpecificationBase >( [&] ( FieldSpecificationBase const & fs )
+  {
+    localIndex isFieldNameFound = 0;
+    // map from set name to a flag (1 if targetSet not empty, 0 otherwise)
+    map< string, localIndex > isTargetSetNonEmpty;
+    // map from set name to a flag (1 if targetSet has been created, 0 otherwise)
+    map< string, localIndex > isTargetSetCreated;
+
+    // Step 1: collect all the set names in a map (this is made necessary by the "apply" loop pattern
+
+    array1d< string > const & setNames = fs.getSetNames();
+    for( localIndex i = 0; i < setNames.size(); ++i )
+    {
+      isTargetSetNonEmpty[setNames[i]] = 0;
+      isTargetSetCreated[setNames[i]] = 0;
+    }
+
+    // Step 2: apply the boundary condition
+
+    fs.apply< dataRepository::Group >( mesh,
+                                       [&]( FieldSpecificationBase const &,
+                                            string const & setName,
+                                            SortedArrayView< localIndex const > const & targetSet,
+                                            Group & targetGroup,
+                                            string const fieldName )
+    {
+      dataRepository::InputFlags const flag = fs.getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).getInputFlag();
+
+      // 2.a) If we enter this loop, we know that the set has been created
+      //      Fracture/fault sets are created later and the "apply" call silently ignores them
+      isTargetSetCreated.at( setName ) = 1;
+
+      // 2.b) If the fieldName is registered on this target, we record it
+      //      Unfortunately, we need two exceptions:
+      //       - FieldSpecification that do not target a field, like Aquifer, Traction, Equilibrium, etc. For these, the check is not
+      // necessary (the user cannot mess up)
+      //       - Face boundary conditions that target cell-based quantities, like the face BC of the flow solvers
+      if( targetGroup.hasWrapper( fieldName ) ||
+          flag == InputFlags::FALSE || // no need to check input if the input flag is false (Aquifer, Traction, Equilibrium do not target a
+                                       // field)
+          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() ) // the field names of the face BCs are not always
+                                                                                     // registered on
+                                                                                     // the faceManager...
+      {
+        isFieldNameFound = 1;
```

Comment:
It's a personal preference, but even if you're dealing with `integers` instead of `bool`, I'd stick to `true/false` when it comes to assigning.
```suggestion
        isFieldNameFound = true;
```
Still you to decide.

Reply:
Here, I left the code as is because of the limitations of our `MpiWrapper`s with `bool`.